lib/utility/_query.js

```diff
@@ -13,7 +13,7 @@ exports._query = function _query(query, data, options, callback, executor) {
 		callback = options;
 
 		// Checks if data contains options
-		if (data && data.useResultSet) {
+		if (data && (data.useResultSet || data.write)) {
```

Comment:
Yes, I also thought so, but the 'data' can be array or object, which makes this detection not possible with current the implementation.